[quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37761 [quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d**
* #37759 [quant] quantized::conv1d_prepack, quantized::conv_transpose1d_prepack, and quantized::conv_transpose1d
* #37711 [quant] quantized::conv_transpose2d
* #37777 [quant] quantized::conv_transpose2d_prepack
* #37497 [quant] QNNPACK factoring out common structs from conv/deconv
* #36790 [quant] QNNPACK deconv kernel and tests
* #37405 [quant] QNNPACK deconvolution packing



Differential Revision: [D21379145](https://our.internmc.facebook.com/intern/diff/D21379145)

Commit:
[quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d

[ghstack-poisoned]

Commit:
Update on "[quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d"




Differential Revision: [D21379145](https://our.internmc.facebook.com/intern/diff/D21379145)

[ghstack-poisoned]

jerryzh168 commented on torch/nn/quantized/modules/conv.py:
```diff
@@ -524,10 +524,237 @@ def __init__(self, in_channels, out_channels, kernel_size, stride,
+        # https://github.com/pytorch/pytorch/issues/23890
+        if len(input.shape) != 3:
+            raise ValueError("Input shape must be `(N, C, L)`!")
+        return ops.quantized.conv_transpose1d(
+            input, self._packed_params,
```
btw, we have a refactor: https://github.com/pytorch/pytorch/pull/35923 that I'm trying to land right now, please change the API accordingly after that(remove self.stride - self.groups)

Commit:
Update on "[quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d"




Differential Revision: [D21379145](https://our.internmc.facebook.com/intern/diff/D21379145)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d"




Differential Revision: [D21379145](https://our.internmc.facebook.com/intern/diff/D21379145)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d"




Differential Revision: [D21379145](https://our.internmc.facebook.com/intern/diff/D21379145)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d"




Differential Revision: [D21379145](https://our.internmc.facebook.com/intern/diff/D21379145)

[ghstack-poisoned]

Commit:
Update on "[quant] quantized.ConvTranspose1d and quantized.ConvTranspose2d"




Differential Revision: [D21379145](https://our.internmc.facebook.com/intern/diff/D21379145)

[ghstack-poisoned]
